Assets/GameAssets/Player/PlayerScripts/PlayerHealth.cs

```diff
@@ -53,9 +53,23 @@ public override void TakeDamage(float damage, IBaseDamage damageSource)
 
             RuntimeSfxManager.APlayOneShotSfx?
                 .Invoke(FmodSfxEvents.Instance.PlayerTookDamage, gameObject.transform.position);
+
+            _iFrameCoroutine = StartCoroutine(InvincibilityFrames());
         }
     }
 
+    /// <summary>
+    /// used to grant the player temporary invincibility after taking damage
+    /// </summary>
+    /// <returns></returns>
+    private IEnumerator InvincibilityFrames()
+    {
+        _shouldTakeDamage = false;
+        yield return new WaitForSeconds(_iFrameDelayInSeconds);
+        _shouldTakeDamage = true;
```

Comment:
After the coroutine is over make sure to set the _iFrameCoroutine to null. Otherwise it remains stored even though its not in use